### PR TITLE
[SDK/CLI] Log metrics for local to cloud run

### PR DESCRIPTION
**Pull request by 0mza987**
# Description

Log 2 types of metrics:
- User metrics that when user uses `loge_metric` in code.
- System metrics that starts with `__pf__`

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by zhengfeiwang — APPROVED** [review]

File: src/promptflow-devkit/promptflow/_sdk/_orchestrator/run_submitter.py [hunk 1]
```diff
@@ -212,8 +213,19 @@ def _submit_bulk_run(
...
             # exceptions
             local_storage.dump_exception(exception=exception, batch_result=batch_result)
-            # system metrics: token related
-            system_metrics = batch_result.system_metrics.to_dict() if batch_result else {}
+            # system metrics
+            system_metrics = {}
+            if batch_result:
+                system_metrics.update(batch_result.system_metrics.to_dict())  # token related
+                system_metrics.update(
+                    {f"{SystemMetricKeys.NODE_PREFIX}.{k}": v for k, v in batch_result.node_status.items()}
```
Comment (D-W-) [inline src/promptflow-devkit/promptflow/_sdk/_orchestrator/run_submitter.py:221]: Will flex flow have this metric?

File: src/promptflow-azure/promptflow/azure/operations/_async_run_uploader.py [hunk 2]
```diff
@@ -242,6 +235,33 @@ async def _upload_instance_results(self) -> str:
...
+
+        # convert metrics to float values
+        try:
+            metrics = {k: float(v) for k, v in metrics.items()}
+        except Exception as e:
+            raise UserErrorException(f"Failed to convert metrics {metrics!r} to float values. Error: {e}") from e
+
+        # write metrics to metric service
+        for k, v in metrics.items():
+            await self.metric_client.log_metric(self.run.name, k, v)
```
Comment (D-W-) [inline src/promptflow-azure/promptflow/azure/operations/_async_run_uploader.py:262]: Can we log in async to make it quicker?

**Review by crazygao — APPROVED** [review]

File: src/promptflow-devkit/promptflow/_sdk/_orchestrator/run_submitter.py (hunk 1 above)
Comment (0mza987) [reply src/promptflow-devkit/promptflow/_sdk/_orchestrator/run_submitter.py:221]: Yes, as long as its metrics are stored in the `batch_result`, if not it should be a bug of batch engine.

File: src/promptflow-azure/promptflow/azure/operations/_async_run_uploader.py (hunk 2 above)
Comment (0mza987) [reply src/promptflow-azure/promptflow/azure/operations/_async_run_uploader.py:262]: This is already an async function call? Do you mean call metrics async endpoint inside the metric client? Can sync it after the holiday tho.

**Pull request merged**